src/benchmark.rs

```diff
@@ -0,0 +1,74 @@
+//! Benchmark features to implement different specific benchmarks.
+//!
+//! This is useful for creating different benchmark on time of execution or
+//! the number of instruction for example.
+
+use core::arch::asm;
+
+use crate::arch::{Arch, Architecture, Csr};
+
+pub struct Benchmark {
+    // Does the benchmark started.
+    pub is_running: bool,
+
+    // Temporary value to store previous state (e.g. state when the benchmark started to compare).
+    pub previous_instr_count: usize,
+
+    pub previous_timer: usize,
+}
+
+impl Benchmark {
+    #[allow(dead_code)]
```

Comment:
Hmmm why do you need `#[allow(dead_code)]` here? It seems like it is used?

Reply:
It is because I used macro to enable code or not before, so the code wasn't used sometime and this disabled the warning. Now I don't need this anymore, good catch thanks :)